pkg/ingester/ingester.go

```diff
@@ -2203,7 +2203,8 @@ const queryStreamBatchMessageSize = 1 * 1024 * 1024
 func (i *Ingester) QueryStream(req *client.QueryRequest, stream client.Ingester_QueryStreamServer) (err error) {
 	defer func() { err = i.mapReadErrorToErrorWithStatus(err) }()
 
-	spanlog, ctx := spanlogger.New(stream.Context(), i.logger, tracer, "Ingester.QueryStream")
+	ctx := stream.Context()
+	spanlog := spanlogger.FromContext(ctx, i.logger)
 	defer spanlog.Finish()
```

Comment:
do we still want to finish this span here? wouldn't the parent span be finished?

Reply:
+1

Reply:
🤦 Good catch, I've fixed this in #11726.